.github/workflows/check-manifests-changes.yml

```diff
@@ -0,0 +1,128 @@
+  MAIN_CACHE_KEY: main-manifests-${{ github.run_id }}-${{ github.run_attempt }}
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+jobs:
+  create-pr-manifests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout PR branch
+        uses: actions/checkout@v4
+
+      - name: Run 'make manifests' on PR branch
+        run: |
+          make dry-run-control-plane
+          mkdir -p ./cache/pr
+          mv ./dry-run/manifests.yaml ./cache/pr/manifests.yaml
+
+      - name: Save PR manifests in cache
+        uses: actions/cache@v3
+        with:
+          path: ./cache/pr/
+          key: ${{ env.PR_CACHE_KEY }}
+
+  create-main-manifests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout main branch
+        uses: actions/checkout@v4
+        with:
+          ref: main
+
+      - name: Run 'make manifests' on main branch
+        run: |
+          make dry-run-control-plane
+          mkdir -p ./cache/main
+          mv ./dry-run/manifests.yaml ./cache/main/manifests.yaml
+
+      - name: Save main manifests in cache
+        uses: actions/cache@v3
+        with:
+          path: ./cache/main/
+          key: ${{ env.MAIN_CACHE_KEY }}
+
+  diff-manifests:
+    needs:
+      - create-pr-manifests
+      - create-main-manifests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Restore PR manifests from cache
+        uses: actions/cache@v3
+        with:
+          path: ./cache/pr/
+          key: ${{ env.PR_CACHE_KEY }}
+
+      - name: Restore main manifests from cache
+        uses: actions/cache@v3
+        with:
+          path: ./cache/main/
+          key: ${{ env.MAIN_CACHE_KEY }}
+
+      - name: Compare Manifests
+        id: compare-manifests
+        run: |
+          set +e
+          DIFF_OUTPUT=$(diff ./cache/pr/manifests.yaml ./cache/main/manifests.yaml)
+          EXIT_CODE=$?
+          if [[ $EXIT_CODE != 0 ]]; then
+            echo "❌ Detected diff in manifests!"
+            echo "$DIFF_OUTPUT"
+            echo "outdated_manifests=true" >> $GITHUB_OUTPUT
+            exit $EXIT_CODE
+          fi
+          echo "✅ No diff in manifests, all good."
+          echo "outdated_manifests=false" >> $GITHUB_OUTPUT
+
+      - name: Add PR Comment if Manifests Are Outdated
+        if: steps.compare-manifests.outputs.outdated_manifests == 'true'
+        uses: actions/github-script@v7
+        with:
+          script: |
+            github.rest.issues.createComment({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              issue_number: context.payload.pull_request.number,
+              body: "❌ **Detected diff in manifests!** Run 'make manifests' and commit changes."
+            });
+            github.rest.issues.addLabels({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              issue_number: context.payload.pull_request.number,
+              labels: ["outdated-manifests"]
+            });
```

Comment:
The check above (compare-manifests) doesn't determine if the manifests are outdated. It determines if there is a diff between main and PR which may be an intended thing. Please adjust the descriptions and variable names accordingly.